cabal-install-solver/src/Distribution/Solver/Modular/Message.hs

```diff
@@ -14,26 +14,51 @@ import qualified Data.Set as S
 import Data.Maybe (catMaybes, mapMaybe, isJust)
 import Prelude hiding (pi)
 
-import Distribution.Pretty (prettyShow) -- from Cabal
+import Distribution.Pretty ( prettyShow ) -- from Cabal
 
 import qualified Distribution.Solver.Modular.ConflictSet as CS
 import Distribution.Solver.Modular.Dependency
-import Distribution.Solver.Modular.Flag ( QFN, QSN )
-import qualified Distribution.Solver.Modular.Flag as Flag ( showQFN, showQFNBool, showQSN, showQSNBool )
+    ( Var(P),
+      ConflictSet,
+      showConflictSet,
+      QGoalReason,
+      GoalReason(DependencyGoal, UserGoal),
+      Goal(Goal),
+      DependencyReason(DependencyReason),
+      ExposedComponent(..),
+      PkgComponent(PkgComponent),
+      CI(Constrained, Fixed),
+      showDependencyReason )
+import Distribution.Solver.Modular.Flag
+    ( QSN, QFN, showQFNBool, showQSNBool, showQFN, showQSN )
 import Distribution.Solver.Modular.MessageUtils
-         (showUnsupportedExtension, showUnsupportedLanguage)
+    ( showUnsupportedExtension, showUnsupportedLanguage )
 import Distribution.Solver.Modular.Package
+    ( PI(PI), showI, showPI )
 import Distribution.Solver.Modular.Tree
-         ( FailReason(..), POption(..), ConflictingDep(..) )
+    ( FailReason(..), POption(..), ConflictingDep(..) )
 import Distribution.Solver.Modular.Version
+    ( VR, Ver, showVer, showVR, (.||.) )
+
 import Distribution.Solver.Types.ConstraintSource
+    ( ConstraintSource (..), showConstraintSource )
 import Distribution.Solver.Types.PackagePath
+    ( QPN, Qualified(Q), showQPN )
 import Distribution.Solver.Types.Progress
-import Distribution.Solver.Types.ProjectConfigPath (docProjectConfigPathFailReason)
+    ( Progress(..),
+      SummarizedMessage(..),
+      EntryAtLevel(..),
+      Entry(..) )
+import Distribution.Solver.Types.ProjectConfigPath
+    ( docProjectConfigPathFailReason)
 import Distribution.Types.LibraryName
+    ( LibraryName(LSubLibName, LMainLibName) )
 import Distribution.Types.UnqualComponentName
-import Text.PrettyPrint (nest, render)
+    ( unUnqualComponentName )
+
+import Text.PrettyPrint ( nest, render )
 
+-- A data type to hold state information for the modular solver.
 data Message =
```

Comment:
Similarly to my previous comment, I think that it is more accurate to describe this type as a log message than solver state.